src/librustdoc/visit_ast.rs

```diff
@@ -401,30 +401,23 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
                 };
                 om.traits.push(t);
             }
-            hir::ItemKind::Impl {
-                unsafety,
-                polarity,
-                defaultness,
-                constness,
-                defaultness_span: _,
-                ref generics,
-                ref of_trait,
-                self_ty,
-                ref items,
-            } => {
+            hir::ItemKind::Impl(ref impl_) => {
                 // Don't duplicate impls when inlining or if it's implementing a trait, we'll pick
                 // them up regardless of where they're located.
-                if !self.inlining && of_trait.is_none() {
-                    let items =
-                        items.iter().map(|item| self.cx.tcx.hir().impl_item(item.id)).collect();
+                if !self.inlining && impl_.of_trait.is_none() {
+                    let items = impl_
+                        .items
+                        .iter()
+                        .map(|item| self.cx.tcx.hir().impl_item(item.id))
+                        .collect();
                     let i = Impl {
-                        unsafety,
-                        polarity,
-                        defaultness,
-                        constness,
-                        generics,
-                        trait_: of_trait,
-                        for_: self_ty,
+                        unsafety: impl_.unsafety,
+                        polarity: impl_.polarity,
+                        defaultness: impl_.defaultness,
+                        constness: impl_.constness,
+                        generics: &impl_.generics,
+                        trait_: &impl_.of_trait,
+                        for_: &impl_.self_ty,
```

Comment:
I think that this bit would be slightly nicer by using destructuring as it was before by using `hir::ItemKind::Impl(hir::Impl { .. }) =>` like you do in other places.

Reply:
This whole block of code disappeared when rebasing over https://github.com/rust-lang/rust/pull/79312; the new changes are in https://github.com/rust-lang/rust/pull/79322/files#diff-384affc1b4190940f114f3fcebbf969e7e18657a71ef9001da6b223a036687d9R2037. I think you'll agree the new code is a lot nicer :)